### PR TITLE
Introduce MaxQuantRun datastructure

### PR DESCRIPTION
**Pull request by Steffengreiner**
**Description of changes**
Introduces a dedicated `maxQuantRunResult` datastructure containing the result of a parsed maxQuant filetree. 
It's structured similiary to the `nfCorePipelineResult `datastructure and is based on the [ER diagram]. (https://app.diagrams.net/#G1umQLaB6kDNFmZwkdARoc9K_usgHNf35s)

This PR achieves the following: 

1. All Datafiles are properties of the `maxQuantRunResult` datastructure 
2. The MaxQuantRun folder is removed since it's functionality is contained in the datastructure. 
2. Valid and Invalid Json Files are introduced to test the data structure generation
3. The `SampleIds` DataFile is moved outside of its original NFCore Context into a general section, since it's also employed in the maxQuant Dataset. For this reason the` NFCorePipelineResult `and its associated tests have to be adapted as well.

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/datasets/MaxQuantRunResult.groovy [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    private Peptides peptides
+
+    private ProteinGroups proteinGroups
+
+    private RunParameters runParameters
+
+    private Summary summary
+
+    private SampleIds sampleIds
+
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/datasets/MaxQuantRunResult.groovy:58]: is there a reason that these fields are mutable?

File: src/test/resources/examples/resultset/maxquant/valid-resultset-example.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "runParameters": {
+      "name": "mqpar.xml",
+      "fileType": "xml",
+      "path": "./maxquant_run/mqpar.xml"
+    },
+    "summary": {
+      "name": "summary_1234.pdf",
+      "fileType": "pdf",
+      "path": "./maxquant_run/summary_1234.pdf"
```
Comment (sven1103) [inline src/test/resources/examples/resultset/maxquant/valid-resultset-example.json:40]: the correct path must be `./combined/txt/summary_1234.pdf`

**Review by sven1103 — CHANGES_REQUESTED** [review]
Only minor comments

File: src/test/resources/examples/resultset/maxquant/valid-resultset-example.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    "proteinGroups": {
+      "name": "proteinGroups.txt",
+      "fileType": "txt",
+      "path": "./combined/txt/proteinGroups.txt"
+    },
+    "runParameters": {
+      "name": "mqpar.xml",
+      "fileType": "xml",
+      "path": "./maxquant_run/mqpar.xml"
```
[the remark this answers is not in the document]
Comment (Steffengreiner) [reply src/test/resources/examples/resultset/maxquant/valid-resultset-example.json]: True!

File: src/main/groovy/life/qbic/datamodel/datasets/MaxQuantRunResult.groovy (hunk 1 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/datasets/MaxQuantRunResult.groovy]: The fields are similiar to the ones defined in the NFCorePipelineResult Fields but you're right these can be set to final. Nice Catch! 👀

**Review by sven1103 — APPROVED** [review]

**Comment by sven1103** [comment]
[MaxQuant: Write ETL max quant registry class in java openBiS dropbox](https://trello.com/c/U4anpPWL/854-maxquant-write-etl-max-quant-registry-class-in-java-openbis-dropbox)

**Pull request merged**